### PR TITLE
adding node label based affinities for more cost efficient scheduling

### PR DESCRIPTION
**Pull request by oz-r**

### CODE REVIEW

File: execution/adapter/eks_adapter.go [hunk 1]
```diff
@@ -216,6 +216,39 @@ func (a *eksAdapter) constructAffinity(executable state.Executable, run state.Ru
 			Operator: corev1.NodeSelectorOpNotIn,
 			Values:   gpuNodeTypes,
 		})
+		//adding node group affinities for non-gpu runs
+		if (executableResources.Gpu == nil || *executableResources.Gpu <= 0) && (run.Gpu == nil || *run.Gpu <= 0) {
+			requiredMatch = append(requiredMatch, corev1.NodeSelectorRequirement{
+				Key:      "beta.kubernetes.io/instance-type",
```
Comment (zhangluva) [inline execution/adapter/eks_adapter.go:222]: I think the key should be `sfix/instance.size`

**Review by zhangluva — COMMENTED** [review]
Generally, the more we try to control, the more conflicts it may generate. Our focus is to avoid the small jobs on large instances and cause the life of large instances extended without enough load. So we probably just need to ensure the small ones are scheduled on the small instances, at least to try it out first.

**Comment by oz-r** [comment]
> Generally, the more we try to control, the more conflicts it may generate. Our focus is to avoid the small jobs on large instances and cause the life of large instances extended without enough load. So we probably just need to ensure the small ones are scheduled on the small instances, at least to try it out first.

Ok, I'll remove the constraints for everything but the `small` size.

**Review by zhangluva — APPROVED** [review]
LGTM. Please apply the other PR first, make sure the labels are there and then apply this

**Pull request merged**